### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by bvssvni**
DO NOT MERGE YET!

### CODE REVIEW

**Comment by bvssvni** [comment]
@kvark I don't get any cube in the "cube" example. Do have any idea of what's wrong?

Also, getting this in the "paint" example when trying to paint (seems related to https://github.com/gfx-rs/gfx/pull/929):

```
$ RUST_BACKTRACE=1 cargo run --bin paint
     Running `target/debug/paint`
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: OutOfBounds { target: [300, 300, 1], source: [300, 300, 0] }', ../src/libcore/result.rs:746
stack backtrace:
   1:        0x1011c17e8 - sys::backtrace::tracing::imp::write::hdcc91e23d52f1e9feuv
   2:        0x1011c3205 - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.44401
   3:        0x1011c2e28 - panicking::default_handler::h5bfcd298419b6ef54Wz
   4:        0x1011bda86 - sys_common::unwind::begin_unwind_inner::hbee05b49f1a706c20su
   5:        0x1011bdf3e - sys_common::unwind::begin_unwind_fmt::hac9a85f022410ea86ru
   6:        0x1011c0d07 - rust_begin_unwind
   7:        0x1011e70e0 - panicking::panic_fmt::h98dcb863ad76e04b7mM
   8:        0x1010b151c - result::unwrap_failed::h9097994666915507250
   9:        0x1010b13b9 - result::Result<T, E>::unwrap::h16174230633879458711
  10:        0x101064ad3 - main::h694ec353376c3668kaa
  11:        0x1011c2a32 - sys_common::unwind::try::try_fn::h14955010582267137138
  12:        0x1011c0c9b - __rust_try
  13:        0x1011c28f0 - rt::lang_start::hb0216a17e8d25b77aPz
  14:        0x1010b3589 - main
Process didn't exit successfully: `target/debug/paint` (exit code: 101)
```

**Comment by kvark** [comment]
Yes, it does look like #929. I assume you didn't use gfx from master?

**Comment by bvssvni** [comment]
Getting another error on Gfx master for the "paint" example:

```
$ RUST_BACKTRACE=1 cargo run --bin paint
   Compiling piston2d-gfx_graphics v0.26.0
   Compiling piston_window v0.41.0
   Compiling piston-examples v0.0.0 (file:///Users/sven/rust/piston-examples)
     Running `target/debug/paint`
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: UnitCountMismatch { target: 0, slice: 90000 }', ../src/libcore/result.rs:746
stack backtrace:
   1:        0x108fdcfa8 - sys::backtrace::tracing::imp::write::hdcc91e23d52f1e9feuv
   2:        0x108fde9c5 - panicking::default_handler::_$u7b$$u7b$closure$u7d$$u7d$::closure.44401
   3:        0x108fde5e8 - panicking::default_handler::h5bfcd298419b6ef54Wz
   4:        0x108fd9246 - sys_common::unwind::begin_unwind_inner::hbee05b49f1a706c20su
   5:        0x108fd96fe - sys_common::unwind::begin_unwind_fmt::hac9a85f022410ea86ru
   6:        0x108fdc4c7 - rust_begin_unwind
   7:        0x1090028a0 - panicking::panic_fmt::h98dcb863ad76e04b7mM
   8:        0x108ecb81c - result::unwrap_failed::h4908296651657540315
   9:        0x108ecb6b9 - result::Result<T, E>::unwrap::h11578594368143417584
  10:        0x108e7edd3 - main::h694ec353376c3668kaa
  11:        0x108fde1f2 - sys_common::unwind::try::try_fn::h14955010582267137138
  12:        0x108fdc45b - __rust_try
  13:        0x108fde0b0 - rt::lang_start::hb0216a17e8d25b77aPz
  14:        0x108ecd889 - main
Process didn't exit successfully: `target/debug/paint` (exit code: 101)
```

**Comment by kvark** [comment]
@bvssvni thank you for the report. Looks like `ImageInfo` returns 0 for texel count, and that's a bug in GFX. Possibly broken by #929... I'll look into it.

**Comment by sectopod** [comment]
I've just publishes [gfx_core](https://crates.io/crates/gfx_core) v0.2.1 with the relevant fixes. It should be picked up after doing `cargo update` automatically. Hope this works for you.

**Comment by bvssvni** [comment]
@sectopod Thanks! The paint example works now.

**Comment by kvark** [comment]
@sectopod Sweet, I'm glad this issue is resolved.

**Comment by bvssvni** [comment]
@kvark Now it's just the mysterious missing cube left.

**Comment by bvssvni** [comment]
@kvark I'm running out of hypothesis to test. Suggestions?

**Comment by kvark** [comment]
@bvssvni can I repro? I tried grabbing this PR yesterday but it didn't build. Will try again.

File: src/paint.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ fn main() {
                         let new_y = (last_y + (diff_y * delta)) as u32;
                         if new_x < width && new_y < height {
                             canvas.put_pixel(new_x, new_y, im::Rgba([0, 0, 0, 255]));
-                            texture.update(&mut*e.factory.borrow_mut(), &canvas).unwrap();
+                            texture.update(&mut*e.encoder.borrow_mut(), &canvas).unwrap();
```
Comment (crazymykl) [inline src/paint.rs]: Does this still need to be `&mut*`, or would `&mut` be sufficient?

Comment (bvssvni) [inline src/paint.rs]: @crazymykl `&mut` is sufficient.

**Comment by bvssvni** [comment]
Here is the apitrace for the missing cube:

[cube.trace.zip](https://github.com/PistonDevelopers/piston/files/215306/cube.trace.zip)

**Comment by bvssvni** [comment]
I'll merge this an open up a bug issue for the missing cube.

**Comment by kvark** [comment]
@bvssvni I see one issue that could explain the missing cube. `I8Scaled` is no more, and we are just passing in integers. But the shader code wasn't update to take that into account. Should be:
```glsl
in ivec3 a_pos; // instead of vec3
in ivec2 a_tex_coord; // instead of vec2
```

I fully admit that it's our fault at GFX for not erroring out properly on that mismatch. Since PSOs were introduced the error detection was left in a slightly worse state than it was, and needs to be improved.

**Comment by bvssvni** [comment]
@kvark Thanks! Fixed in https://github.com/PistonDevelopers/piston-examples/pull/341.

**Pull request merged**